modules/transactions/src/main/java/org/apache/ignite/internal/tx/InternalTransaction.java

```diff
@@ -105,4 +105,15 @@ public interface InternalTransaction extends Transaction {
      * provided no transaction for an operation) or not.
      */
     boolean implicit();
+
+    /**
+     * Finishes a read-only transaction with a specific execution timestamp.
+     *
+     * @param commit Commit flag. The flag is ignored for read-only transactions.
+     * @param executionTimestamp The timestamp is the time when a read-only transaction is applied to the remote node.
```

Comment:
It's not a good idea to mention read-only specifics in common InternalTransacion. It also worth to mention, that executionTimestamp is equals or **reasonably greater** to the time when a read-only transactions **read timestamp**.

Reply:
Unfortunately, we already have methods for specific transaction types in this interface.